src/App.tsx

```diff
@@ -1,26 +1,114 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from 'react';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
 
-const USER_ID = 0;
+export enum ErrorType {
+  TodosLoad = 'Unable to load todos',
+  EmptyTitle = 'Title should not be empty',
+  UnableToAddTodo = 'Unable to add a todo',
+  UnableToDeleteTodo = 'Unable to delete a todo',
+  UnableToUpdateTodo = 'Unable to update a todo',
+}
+
+export enum Filter {
+  all = 'All',
+  active = 'Active',
+  completed = 'Completed',
+}
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [isTodoEditing, setIsTodoEditing] = useState(false);
+  const [selectedPostId, setSelectedPostId] = useState(0);
+  const [currentError, setCurrentError] = useState<ErrorType | ''>('');
+  const [selectedFilter, setSelectedFilter] = useState(Filter.all);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isDeleteAllPressed, setIsDeleteAllPressed] = useState(false);
+  const [isToggleAllPressed, setIsToggleAllPressed] = useState(false);
```

Comment:
About isTodoEditing state can know only TodoItem component.
some states looks like redundant (selectedPostId, isLoading, isDeleteAllPressed, isToggleAllPressed).
Try to simplify logic and avoid them.
  
```
  const [currentError, setCurrentError] = useState<ErrorType | ''>('');
  const [selectedFilter, setSelectedFilter] = useState(Filter.all);
  const [todos, setTodos] = useState<Todo[]>([]);
  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
```
select